src/CoreBundle/CronJob/CronjobEnablingService.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+/*
+ * This file is part of the Chameleon System (https://www.chameleonsystem.com).
+ *
+ * (c) ESONO AG (https://www.esono.de)
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ChameleonSystem\CoreBundle\CronJob;
+
+use ChameleonSystem\CoreBundle\Exception\CronjobEnableException;
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\DBALException;
+use esono\pkgCmsCache\CacheInterface;
+use TdbCmsConfig;
+
+class CronjobEnablingService implements CronjobEnablingServiceInterface
+{
+    /**
+     * @var Connection
+     */
+    private $connection;
+
+    /**
+     * @var CacheInterface
+     */
+    private $cache;
+
+    public function __construct(Connection $connection, CacheInterface $cache)
+    {
+        $this->connection = $connection;
+        $this->cache = $cache;
+    }
+
+    public function isCronjobExecutionEnabled(): bool
+    {
+        $config = TdbCmsConfig::GetInstance(true);
```

Comment:
Why do we need to reload?

Reply:
Will this change if this flag was changed externally during execution of all cronjobs?

Reply:
Not sure if I understand completely; in general we should not need to think about cache freshness on simple reads. If any code changes the value, it is the responsibility of that code to call cache triggers. We do this in this service, and for normal changes in the table editor, the system will take care.

Reply:
ok removed

The new methods in CronjobEnablingServiceInterface for enabling and isabling do a cache triggering.